### ISSUE TITLE
`c is not a function`

### ISSUE THREAD

**Issue by TechStudent10** [pocketbase/js-sdk#120]
I've written my own state-based router for an Electron+React app that I'm building. The app initially starts on the user page:

![image](https://user-images.githubusercontent.com/76978184/209606763-e72188b0-701d-40c7-84d8-a21e37282dd9.png)

Clicking on `Dashboard`, it works as intended:

![image](https://user-images.githubusercontent.com/76978184/209606801-d55e8c31-99d3-4eee-9f8f-fc7aee828f43.png)

But then switching back to the user page raises this error:

![image](https://user-images.githubusercontent.com/76978184/209606844-120c932e-22f9-4000-97e1-f0f85ff13fb4.png)

`TypeError: c is not a function` and `Uncaught TypeError: c is not a function`

**Comment by ganigeorgiev** [comment]
Without a code sample what you are doing I'm not sure how to help.

I don't think the error is related to the SDK. From the stack trace it indicates that it comes from some minified js files. You can try inserting debug breakpoints from the browser insect tool and see where it fail.

The request could throw an error (but not "c is not a function") in case of auto cancellation. If you haven't already, you can check https://github.com/pocketbase/js-sdk#auto-cancellation.

**Comment by TechStudent10** [comment]
Alright I'll try all your suggestions. If you still want it, here is code sample:

```jsx
import React, { useState } from 'react';

export default function Login(props) {
    const [username, setUsername] = useState("")
    const [password, setPassword] = useState("")
    const [passwordConfirm, setPasswordConfirm] = useState("")
    const [email, setEmail] = useState("")
    const [loginPage, setLoginPage] = useState(true)

    const login = async () => {
        await props.pb.collection("users").authWithPassword(username, password)
    };
    const signup = async () => {
        const data = {
            username,
            password,
            passwordConfirm: passwordConfirm
        }
        const createdUser = await props.pb.collection("users").create(data)
        await login()
    };
    const logout = () => {
        props.pb.authStore.clear()
    };
...
```
The functions are called below in the HTML.

**Comment by ganigeorgiev** [comment]
@TechStudent10 Based only on the above snippet, it should work fine if `props.pb` is a nonempty PocketBase JS SDK instance.

You can also wrap you `await` calls with `try/catch` to prevent propagating the error so that you can debug it more easily.

**Comment by TechStudent10** [comment]
Okay I've located the issue. After accidentally figuring out how to switch from production React (minified) to development (non-minified), I get a more detailed error saying the function provided at the end of the component life-cycle isn't a function. 

TL;DR I thought `pb.authStore.onChange` returned a function, which it doesn't (apparently), so React is trying to call it but it can't because it's is getting `undefined`, which obviously isn't callable.

**Comment by TechStudent10** [comment]
That and the fact that you can't use an `async` function in a `useEffect` (different component):

![image](https://user-images.githubusercontent.com/76978184/209700893-42de1022-b2a4-49bb-980a-c6a09dc21378.png)
